src/app/login/login.component.ts

```diff
@@ -12,19 +13,31 @@ export class LoginComponent implements OnInit {
   password: any = '';
   showFailedLoginInfo = false;
 
-  constructor(private router: Router, private authservice: AuthService) {}
+
+  constructor(private router: Router, private authservice: AuthService,private spinner: NgxSpinnerService) {}
 
   ngOnInit() {}
 
   login() {
-    if (this.username && this.password) {
-      console.log('Succesfully logged ' + this.username);
-      this.showFailedLoginInfo = false;
-      this.authservice.login(this.username, this.password);
-      this.router.navigate(['/courses']);
-    } else {
-      console.log('Unsuccesfull... ' + this.username);
-      this.showFailedLoginInfo = true;
-    }
+    this.spinner.show();
+    this.delay(3000).then(any => {
```

Comment:
Use rxjs for this purpose.
Also, the component is not the best place for this. Move it to the service, to the method `this.authservice.login`, which should return observable if a user logged, or throw an error if not.